src/test_util/mod.rs

```diff
@@ -50,6 +50,20 @@ macro_rules! assert_downcast_matches {
     }};
 }
 
+#[macro_export]
+macro_rules! log_event {
```

Comment:
Also curious to get feedback on this addition of this helper macro. I found it useful in tests to easily construct events to assert against.

Reply:
Seems like a reasonable addition to me! I know @Hoverbear has thoughts around event usability as well.